video/image_writer.c

```diff
@@ -262,7 +297,237 @@ static bool write_jpeg(struct image_writer_ctx *ctx, mp_image_t *image, FILE *fp
 
     jpeg_destroy_compress(&cinfo);
 
-    return true;
+    return !fclose(fp);
+}
+
+#endif
+
+#if HAVE_AVIF_MUXER
+
+static void log_side_data(struct image_writer_ctx *ctx, AVPacketSideData *data,
+                          int size)
+{
+    static char dbgbuff[129];
+    if (size)
+        MP_DBG(ctx, "write_avif() packet side data:\n");
+
+    for (int i = 0; i < size; i++) {
+        AVPacketSideData *sd = &data[i];
+        int k = 0;
+        for (; k < MPMIN(sd->size, 64); k++)
+            sprintf(dbgbuff + k*2, "%02x", sd->data[k]);
+        dbgbuff[k] = 0;
```

Comment:
Nitpick: I prefer `'\0'` since to me personally it's clear you're null-terminating a string. Feel free to leave as-is.